tornado-api/src/main/java/uk/ac/manchester/tornado/api/types/arrays/TornadoNativeArray.java

```diff
@@ -33,7 +42,10 @@
  * The constant {@link ARRAY_HEADER} represents the size of the header in bytes.
  * </p>
  */
-public abstract sealed class TornadoNativeArray permits ByteArray, CharArray, DoubleArray, FloatArray, IntArray, LongArray, ShortArray, HalfFloatArray {
+public abstract sealed class TornadoNativeArray //
+        permits ByteArray, CharArray, DoubleArray, FloatArray, HalfFloatArray, //
+        IntArray, LongArray, ShortArray, Tensor, TensorByte, TensorFloat16, //
```

Comment:
The permits should only include the base class for Tensors. And then the Tensor class itself include the tensor subtypes. Is this possible?

Reply:
I did an experiment on that, but it seems that it is not allowed.

I am getting the following error:` Invalid permits clause: 'TensorByte' must directly extend 'TornadoNativeArray'`